packages/nextcloud/test/cookbook_test.dart

```diff
@@ -16,64 +15,31 @@ void main() {
       late DockerContainer container;
       late NextcloudClient client;
 
-      setUpAll(() async {
+      setUp(() async {
         container = await DockerContainer.create(preset);
         client = await TestNextcloudClient.create(container);
-
-        final source = Directory('test/files/Recipes');
-        await client.webdav.mkcol(PathUri.parse(source.path.substring(11)));
-
-        for (final element in source.listSync(followLinks: false, recursive: true)) {
-          // remove the `test/files/` prefix
-          final path = PathUri.parse(element.path.substring(11));
-
-          if (element is Directory) {
-            await client.webdav.mkcol(path);
-          } else if (element is File) {
-            await client.webdav.putFile(
-              element,
-              element.statSync(),
-              path,
-            );
-          }
-        }
-
-        resetFixture();
       });
 
-      tearDownAll(() async {
+      tearDown(() async {
         if (Invoker.current!.liveTest.errors.isNotEmpty) {
           print(await container.allLogs());
         }
         container.destroy();
       });
```

Comment:
It can not reuse the same container for all tests? If you'd just delete all the recipe files after the test is done, shouldn't that reset everything to the initial state?

Reply:
the issue is that I want a fresh predefined set of recipes.
The collection introduced has a lot of nasty "bugs" in them like integer IDs and I want to make sure we can parse all of them. Sometimes this is just testing the upstream php code but I've had quite a few bugs already that make me uncomfortable without these tests.

We could upload all recipes to a different folder and use webdav to recreate the original set :woman_shrugging: 
Is it really such a big performance difference?

Reply:
It depends on the ratio of container startup time vs. test time.
I think you could at least bake the recipes into the docker image if you want to create a new one every time to save some startup time of the test.